lua/autorun/client/cl_cfc_restart_display.lua

```diff
@@ -0,0 +1,71 @@
+local finalTime
+local keyFrame = 0
+local isOpen = false
+local flashState = false
+local CFCRestart = {}
+surface.CreateFont( "CFC_CALIBRI_20PX_DEFAULT", {
+    font = "Calibri",
+    extended = false,
+    size = 40,
+    weight = 500
+})
+
+local function timeLeft()
+    return finalTime - SysTime()
+end
+
+local function restartUI()
+    timer.Create( "CFC_RESTART_FLASHTIMER", 1, 0, function()
+        flashState = not flashState
+    end )
+
+    isOpen = true
+    hook.Remove( "HUDPaint", "CFCDrawRestartAlert" )
+    local delay = net.ReadInt( 16 )
+    local curTime = SysTime()
+    finalTime = curTime + delay
+    hook.Add( "HUDPaint", "CFCDrawRestartAlert", function()
+
+        if isOpen then
+            keyFrame = math.Clamp( keyFrame + 5, 0, ScrW() )
+        else
+            keyFrame = math.Clamp( keyFrame - 10, 0, ScrW() )
+
+            if keyFrame <= 0 then
+                hook.Remove( "HUDPaint", "CFCDrawRestartAlert" )
+            end
+        end
+
+        surface.SetDrawColor( Color( 36, 54, 72 ) )
+        surface.PlaySound( "ambient/alarms/warningbell1.wav" )
+        surface.DrawRect( 0, 5, keyFrame, 50 )
+    
+        CFCRestart.urgencyTime = 30 -- extracting that 30 out from here, should probably be named something better
```

Comment:
These constants should be defined at the top of the file, as they are effectively config values.